Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1566,4 +1566,67 @@ function Set-ToolsRepo
             Write-Host "Successfully set tools-repo on all hosts for datastore $ds_name"
         }
     }
-}
\ No newline at end of file
+}
+
+<#
+.Synopsis
+    Set vSAN compression and deduplication on a cluster or clusters. If deduplication is enabled then compression is required.
+    The default cluster configuration is deduplication and compression but the customer can change that.
+    Choosing neither compression nor deduplication will disable both.
+    This requires action on every physical disk and will take time to complete.
+
+.EXAMPLE
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Compression $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Deduplication $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2"
+
+    Set-vSANCompressDedupe -Cluster "all"
+#>
+
+function Set-vSANCompressDedupe
+{
+    [AVSAttribute(120, UpdatesSDDC = $true)]
```

Comment:
This is actually above current hard cap on the agent, if we really expect 2 hours to complete then we'll need to rollout an agent update.

Reply:
Oh, I thought it had been changed to 3 hours actually. I'll change it to 60 minutes. It might not complete in that time depending on data but the process will have been handed off to vCenter and will complete. Still figuring out a good metric to estimate it.

Reply:
It used to be 3h, we could leave the 2h here, if that's how long it takes and update the cap in the agent at any time later.